CTS1.rst

```diff
@@ -0,0 +1,564 @@
+  }
+
+Smart contract limitations
+==========================
+
+A number of limitations are important to be aware of:
+
+- Smart contract state size is limited to 16 KiB.
+- Smart contract function parameters are limited to 1 KiB.
+- Each logged event is limited to 0.5 KiB.
+- The number of logged events is limited to 64.
+
+.. note::
+
+  Smart contracts where the contract state size limit is to low, can distribute the state across multiple smart contract instances.
+
+Decisions and rationale
+=======================
+
+In this section we point out some of the differences from other popular token standards found on other blockchains, and try to reason why this was decided.
+
+Token ID bytes instead an integer
+---------------------------------
+
+Token standards such as ERC721 and ERC1155 both uses an 256 bit unsigned integer (32 bytes) for the token ID, to support using something like a SHA256 hash for the token ID.
+But in the case where the token ID have no significance other than a simple identifier, smaller sized token IDs can reduce energy costs.
+Which is why we chose to let the first byte indicate the size of the token ID, meaning a token ID can vary between 1 byte and 256 bytes, resulting in more than 10^614 possible token IDs.
+
+Only batched transfers
+----------------------
+
+The specification only have a ``transfer`` smart contract function which takes list of transfer and no function for a single transfer.
+This will result in lower energy cost compared to multiple contract calls and only introduce a small overhead for single transfers.
+The reason for not also including a single transfer function, is to have smaller smart contract modules, which in turn leads to saving cost on every function call.
+
+No token level approval/allowance like in ERC20 and ERC721
+----------------------------------------------------------
+
+This standard only specifies address-level operators and no authentication on per token level.
+The main argument is simplicity and to save energy cost on common cases, but other reasons are:
+
+- A token level authentication requires the token smart contract to track more state, which increases the overall energy cost.
+- For token smart contracts with a lot of token types, such as a smart contract with a large collection of NFTs, a token level authentication could become very expensive.
+- For fungible tokens; approval/allowance introduces an attack vector as `described here<https://docs.google.com/document/d/1YLPtQxZu1UAvO9cZ1O2RPXBbT0mooh4DYKjA_jp-RLM/edit>`.
+
+.. note::
+
+  The specification does not prevent adding more fine-grained authentication, such as a token level authentication.
+
+Receive hook function
+---------------------
+
+The specification requires a token receive hook to be called on a smart contract receiving tokens, this will in some cases prevent mistakes such as sending tokens to smart contracts, which do not defined behavior for receiving tokens.
```

Comment:
```suggestion
The specification requires a token receive hook to be called on a smart contract receiving tokens, this will in some cases prevent mistakes such as sending tokens to smart contracts which do not define behavior for receiving tokens.
```